core/src/main/java/org/springframework/security/access/intercept/AbstractSecurityInterceptor.java

```diff
@@ -375,10 +404,24 @@ public boolean isValidateConfigAttributes() {
 
 	public abstract SecurityMetadataSource obtainSecurityMetadataSource();
 
+	/**
+	 * Sets the {@link AccessDecisionManager} instance.
+	 * @param accessDecisionManager the {@link AccessDecisionManager} instance
+	 * @deprecated use {@link #setAuthorizationManager(AuthorizationManager)} instead
+	 */
+	@Deprecated
 	public void setAccessDecisionManager(AccessDecisionManager accessDecisionManager) {
 		this.accessDecisionManager = accessDecisionManager;
```

Comment:
Agreed.

Reply:
What do you think about `getAccessDecisionManager`?

We can try to cast an `AuthorizationManager` to an `AuthorizationManagerAdapter` and return an `AccessDecisionManager` otherwise throw an exception with message indicating that user should use `getAuthorizationManager` instead.

Reply:
Ideally, `AuthorizationManagerAdapter` would not be public since it perpetuates the usage of a deprecated API. 

It might be possible to make it a private inner class of `AbstractSecurityInterceptor`, in which case you could access its private `AccessDecisionManager`. The configurers would need to know whether to call `setAccessDecisionManager` or `setAuthorizationManager` in that case. The nice thing about that is that we can also know if the application provided an `AccessDecisionManager` and can correctly error if `setAuthorizationManager` is called thereafter.